tests/e2e/scale/test_scale_small_file_workload.py

```diff
@@ -0,0 +1,110 @@
+"""
+Test to exercise Small File Workload
+
+"""
+
+# Builtin modules
+import logging
+import time
+
+# 3ed party modules
+import pytest
+
+# Local modules
+import ocs_ci.ocs.resources.pod as pod
+from ocs_ci.ocs.ripsaw import RipSaw
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import E2ETest, scale
+from ocs_ci.ocs.small_file_workload import smallfile_workload
+
+log = logging.getLogger(__name__)
+
+
+def get_cephfs_data():
+    """
+    Look through ceph pods and find space usage on all cephfilesystem pods
+
+    Returns:
+        Dictionary of byte usage, indexed by pod name.
+    """
+    ceph_pod = pod.get_ceph_tools_pod()
+    ceph_status = ceph_pod.exec_ceph_cmd(ceph_cmd="ceph df")
+    ret_value = {}
+    for pool in ceph_status["pools"]:
+        if "cephfilesystem" in pool["name"]:
+            ret_value[pool["name"]] = pool["stats"]["bytes_used"]
+    return ret_value
+
+
+@pytest.fixture(scope="module")
+def scale_leaks(request):
+    """
+    For scale tests, get bytes used by cephfs at the start,
+    and compare with the cephfs usage at the finish.
+    """
+    pytest.ceph_fs_use = get_cephfs_data()
+
+    def teardown():
+        """
+        After all the ripsaw creation and deltion runs have finished,
+        test to see if there was a leak in storage.
+        """
+        log.info("In scale_leaks teardown")
+        time.sleep(300)
+        orig_data = pytest.ceph_fs_use
+        now_data = get_cephfs_data()
+        for entry in now_data:
+            log.info(f"{entry} now uses {now_data[entry]} bytes")
+            log.info(f"{entry} originally used {orig_data[entry]} bytes")
+            #
+            # Skip metadata check if the space used is still under 1G.
+            # cephfilesytem-metadata pool consumes some storage when
+            # first run, so the first increase is larger than 10M.
+            # Skip metadata check if the space used is still under 1G.
+            #
+            if entry.endswith("metadata") and now_data[entry] < 1000000000:
```

Comment:
10M is a very less number to define a memory leak.

Reply:
What number do you thing would be appropriate here?

Reply:
As discussed offline, let's benchmark the average memory consumption and then check for any leaks after workloads are run.

Reply:
Fixed in latest change.